SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialRPCService.cpp

```diff
@@ -252,14 +252,17 @@ void SpatialRPCService::ExtractRPCsForEntity(Worker_EntityId EntityId, Worker_Co
 	}
 }
 
-void SpatialRPCService::OnCheckoutEntity(Worker_EntityId EntityId)
+void SpatialRPCService::OnCheckoutMulticastRPCComponentOnEntity(Worker_EntityId EntityId)
 {
+	// Precondition: The multicast RPC component exists in the view for the entity with the passed entity ID.
 	const MulticastRPCs* Component = View->GetComponentData<MulticastRPCs>(EntityId);
```

Comment:
Personal preference is to not crash, but log and error if this is incorrect.

Reply:
added an error and return after the check.

Reply:
so will still hard crash in a dev build but will only error in a shipping build (or however build configurations work)

Reply:
The ensure macro is made for this purpose.
`if(ensure(...))
{
 ...
}`

Reply:
that would invert the if statement to work correctly though, right? personal preference is to have if statements like this early out and have the logic of the function at the end, so unless you feel strongly @ImprobableNic I will leave this

Reply:
if(!ensure(Comp != nullptr)) {...}

Reply:
Spoke offline, taking @ImprobableNic 's suggestion